### DISCUSSION TITLE
alt (option) click hiding behavior

### DISCUSSION THREAD

**Discussion by brightpuddle** [asmvik/yabai#1440] [General]
I've used `yabai -m config mouse_modifier alt` for years without any trouble. alt + left click lets me resize windows with the mouse. Suddenly, it's reverted to hiding behavior, i.e. clicking an unfocused window hides all other windows on that desktop, which makes it impossible to use the resize feature.

Everything I've found about this online suggests it's a hardcoded, immutable setting in MacOS. I would believe that except that this just recently stopped working, maybe coinciding with an OS update or yabai update? This is an M1 on 12.6.

Anybody else using alt and seen this behavior? Any ideas?

**Comment by 24unix** [comment]
I don't know if that helps.
I use HyperDock for window movement, no need to click at all.

<img width="756" alt="image" src="https://user-images.githubusercontent.com/36622631/191813967-2061117b-3950-4459-8a69-bd33d1a5ce67.png">

**Comment by asmvik** [comment]
Uh, likely happened in this commit: https://github.com/koekeishiya/yabai/commit/d4aa4c9781dffb741d5348f1a8dbe01ee8047252

Edit: That commit changed the location of where we start tapping events from macOS, because there was a bug where we sometimes didn't detect the correct window, because it is possible to perform a mouse event on a window without the pointer being inside the visible screen rect of the window. 

By changing location of where we start monitoring events, macOS will annotate the events with less/more information, and in this case it populates the window id of the window which is supposed to handle said event.

**Reply by brightpuddle** [reply]
Thanks, @koekeishiya.

Just to clarify it sounds like it's one direction or the other with this, i.e. either the alt + click behavior leaks through and is captured by the OS or there's not enough info in the event to avoid the defect behavior. Is that correct?

Is there any solution that comes to mind with this, or is this the expected behavior now? If the latter, I can find a new mapping.

**Reply by asmvik** [reply]
> Just to clarify it sounds like it's one direction or the other with this, i.e. either the alt + click behavior leaks through and is captured by the OS or there's not enough info in the event to avoid the defect behavior. Is that correct?

Yeah, that is correct. 

> Is there any solution that comes to mind with this, or is this the expected behavior now? If the latter, I can find a new mapping.

For now I think this will become the expected behaviour as it concerns interaction that the user is performing *outside* of yabai, directly with a window through macOS UI controls. There is an open issue to make keybindings for the yabai mouse interactions more flexible, which if/when factored out of yabai specific logic would allow you to keep overriding alt + click and using it to control windows as you have done until now.

My initial though was to do something like `yabai -m window <window_sel> --mouse start_drag` and `yabai -m window <window_sel> --mouse end_drag` or something. Replace <start_drag> with move/resize or whatever operation.

I don't know if that would work in practice; just a thought I had.  I haven't been very interested in adding mouse functionality to *skhd* yet though.
